### PR TITLE
Fixed documentation bugs in DG

### PR DESCRIPTION
**Pull request by EdmundTangg**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -375,6 +379,11 @@ The following sequence diagram details how `FindCommand#execute()` functions.
 The HistoryCommand class is responsible for finding any inflows or outflows for a particular
 PID in the StockPal application.
 
+The history product feature is facilitated by `HistoryCommand` which extends `Command`.
+
+Finding of transactionsis done by calling `ProductList#FindTransactions()`.
```
Comment (Kobot7) [inline docs/DeveloperGuide.md]: Missing space between words: "transactionsis"

File: docs/team/_EdmundTangg.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+**1. New command Feature**
+[#30](https://github.com/AY2324S2-CS2113T-T09-3/tp/pull/30)
+
+What it does:
+This feature added the ability to add a product into the inventory.
+A unique Product ID (PID) is assigned to it as well.
+
+Justification:
+This feature is the core of the application.
+A user needs to be able to add a product in into the application before any further commands can run.
```
Comment (Kobot7) [inline docs/team/_EdmundTangg.md]: Possible grammatical error? "in into..."

File: docs/team/_EdmundTangg.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+What it does:
+This feature added the ability to find a product's name based on a specific keyword.
+
+Justification: 
+This feature is critical to the user's experience as a whole. 
+In the event that the user only remember certain keywords of the product's name, 
+the user can use this command to obtain more information about the product.
+In addition, most commands require the PID of the product to run. 
+Tis command allows the user to obtain the PID of the product that they want
```
Comment (Kobot7) [inline docs/team/_EdmundTangg.md]: Typo ? Assume "Tis" should be "This"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -246,18 +246,18 @@ Aspect: Where to store the individual command details.
     - Code looks very long and repetitive with `FormatUtils.formatCommandDetails(COMMAND_KEYWORD, COMMAND_DESCRIPTION
       , COMMAND_USAGE, COMMAND_FLAGS, COMMAND_FLAG_DESCRIPTIONS)` being called for every single command.
 
-### Add Product Feature
+### New Product Feature
 
 The NewCommand class is responsible for adding a new product to the inventory in the StockPal application.
 
-The add product feature is facilitated by `NewCommand` which extends `Command`.
+The new product feature is facilitated by `NewCommand` which extends `Command`.
```
Comment (NgYaoDong) [inline docs/DeveloperGuide.md:253]: Maybe can say it extends `ListActionCommand` instead

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -357,14 +357,18 @@ currently.
...
 ![DeleteCommandSequenceDiagram](images/DeleteCommandSequence.png)
 
 
 ### Find product feature
 
-The FindCommand class is responsible for finding a new product to the inventory in the StockPal application.
+The FindCommand class is responsible for finding a particular keyword
+in a product's name in the StockPal application.
 
+The find product feature is facilitated by `FindCommand` which extends `Command`.
```
Comment (NgYaoDong) [inline docs/DeveloperGuide.md:369]: Same as `NewCommand`, maybe can say it extends `ListActionCommand` instead

**Pull request merged**